### PR TITLE
add c producer aac audio video sample

### PR DESCRIPTION
**Pull request by chehefen**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+
+PVOID putVideoFrameRoutine(PVOID args)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    PSampleCustomData data = (PSampleCustomData) args;
+    Frame frame;
+    UINT32 fileIndex = 0, frameSize;
+    CHAR filePath[MAX_PATH_LEN + 1];
+    STATUS status;
+    BYTE frameBuffer[200000]; // Assuming this is enough
```
Comment (MushMal) [inline kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c]: Bad assumption. Need to have a proper high-water-mark solution. Also, it's a bad idea to have 200K stack allocation

File: kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    UINT64 runningTime;
+
+    CHK(data != NULL, STATUS_NULL_ARG);
+
+    frame.frameData = frameBuffer;
+    frame.version = FRAME_CURRENT_VERSION;
+    frame.trackId = DEFAULT_VIDEO_TRACK_ID;
+    frame.duration = 0;
+    frame.decodingTs = 0; // relative time mode
```
Comment (MushMal) [inline kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c]: In what sense this is relative mode? Perhaps we should rephrase stating that the time starts from 0?

File: kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        frame.decodingTs = frame.presentationTs;
+        frame.index++;
+
+        fileIndex = (fileIndex + 1) % NUMBER_OF_H264_FRAME_FILES;
+        frame.flags = fileIndex % DEFAULT_KEY_FRAME_INTERVAL == 0 ? FRAME_FLAG_KEY_FRAME : FRAME_FLAG_NONE;
+        SNPRINTF(filePath, MAX_PATH_LEN, "%s/h264SampleFrames/frame-%03d.h264", data->sampleDir, fileIndex + 1);
+        CHK_STATUS(readFrameData(&frame, filePath));
+
+        // synchronize putKinesisVideoFrame to running time
+        runningTime = defaultGetTime() - data->streamStartTime;
```
Comment (MushMal) [inline kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c]: It's hard to understand what is going on and why our "simulated" frame media pipeline needs to synchronized. Suggest we use TimerQueue to fire the frames

File: kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+    BYTE frameBuffer[200000]; // Assuming this is enough
+    UINT64 runningTime;
+
+    CHK(data != NULL, STATUS_NULL_ARG);
+
+    frame.frameData = frameBuffer;
+    frame.version = FRAME_CURRENT_VERSION;
+    frame.trackId = DEFAULT_AUDIO_TRACK_ID;
+    frame.duration = 0;
+    frame.decodingTs = 0; // relative time mode
```
Comment (MushMal) [inline kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c]: Same comments about this fn

File: kinesis-video-c-producer/samples/aacSampleFrames/sample-582.aac [hunk 5]
```diff
@@ -0,0 +1 @@
+!S�_Bp��WL:!Ƹ
```
[the remark this answers is not in the document]
Comment (chehefen) [reply kinesis-video-c-producer/samples/aacSampleFrames/sample-582.aac]: yes we need the audio frames.

File: kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+STATUS readFrameData(PFrame pFrame, PCHAR frameFilePath)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT64 size;
+
+    CHK(pFrame != NULL && frameFilePath != NULL, STATUS_NULL_ARG);
+
+    // Get the size and read into frame
+    CHK_STATUS(readFile(frameFilePath, TRUE, NULL, &size));
+    CHK_STATUS(readFile(frameFilePath, TRUE, pFrame->frameData, &size));
```
[the remark this answers is not in the document]
Comment (chehefen) [reply kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c]: changed to loading all frames into memory during setup.

File: kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c (hunk 1 above)
Comment (chehefen) [reply kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c]: changed to loading all frames into memory during setup.

File: kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c (hunk 2 above)
Comment (chehefen) [reply kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c]: I meant that frames are timestamped relatively.

File: kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c (hunk 3 above)
Comment (chehefen) [reply kinesis-video-c-producer/samples/KvsAacAudioVideoStreamingSample.c]: if we don't synchronize and just rely on thread sleep to simulate "realtime", audio and video timestamp gradually drifts apart and eventually the drift is too large and cause errors.

**Comment by chehefen** [comment]
> Do we have an urgent need for this? May I suggest making the change in the new repository and have the core functionality outsourced just like the samples with webrtc so we can modify once for multiple samples?

sure I can port this to the new repo. However, regarding outsourcing some functionalities I'm more inclined to keep the samples self-contained because it's easier for customer to read.

**Pull request merged**